lib/onix/product.rb

```diff
@@ -549,11 +549,10 @@ def supplies(keep_all_prices_dates=false)
       # add territories if missing
       if @product_supplies
         @product_supplies.each do |ps|
-          if ps.markets && ps.available?
-            if ps.markets.map{|m| m.territory.countries}.flatten.uniq.empty?
-                market_territories = ONIX::Territory.region_to_countries("WORLD")
-            else
-                market_territories = ps.markets.map{|m| m.territory.countries}.flatten.uniq
+          if ps.markets
+            market_territories = ps.markets.map{|m| m.territory.countries}.flatten.uniq
+            if market_territories.empty? && ps.available?
+              market_territories = ONIX::Territory.region_to_countries("WORLD")
```

Comment:
Yes great. Do you have from the last commit, the reason for this special case. I think we need a comment here to explain why.

Reply:
I add a comment